Client/src/main/java/org/gluu/oxauth/client/DeviceAuthzClient.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * oxAuth is available under the MIT License (2008). See http://opensource.org/licenses/MIT for full text.
+ *
+ * Copyright (c) 2014, Gluu
+ */
+
+package org.gluu.oxauth.client;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.gluu.oxauth.model.common.AuthenticationMethod;
+import org.gluu.oxauth.model.util.Util;
+import org.jboss.resteasy.client.ClientExecutor;
+import org.jboss.resteasy.client.ClientRequest;
+import org.json.JSONObject;
+
+import javax.ws.rs.HttpMethod;
+
+import static org.gluu.oxauth.model.authorize.DeviceAuthorizationRequestParam.*;
+import static org.gluu.oxauth.model.authorize.DeviceAuthorizationResponseParam.*;
+
+/**
+ * Encapsulates functionality to make Device Authz request calls to an authorization server via REST Services.
+ */
+public class DeviceAuthzClient extends BaseClient<DeviceAuthzRequest, DeviceAuthzResponse> {
+
+    private static final Logger LOG = Logger.getLogger(DeviceAuthzClient.class);
+
+    /**
+     * Construct a device authz client by providing an URL where the REST service is located.
+     *
+     * @param url The REST service location.
+     */
+    public DeviceAuthzClient(String url) {
+        super(url);
+    }
+
+    @Override
+    public String getHttpMethod() {
+        return HttpMethod.POST;
+    }
+
+    public DeviceAuthzResponse exec() {
+        initClientRequest();
+        return _exec();
+    }
+
+    @Deprecated
+    public DeviceAuthzResponse exec(ClientExecutor clientExecutor) {
+        this.clientRequest = new ClientRequest(getUrl(), clientExecutor);
+        return _exec();
+    }
+
+    private DeviceAuthzResponse _exec() {
+        try {
+            clientRequest.setHttpMethod(getHttpMethod());
+            clientRequest.header("Content-Type", request.getContentType());
+            if (request.getAuthenticationMethod() == AuthenticationMethod.CLIENT_SECRET_BASIC && request.hasCredentials()) {
+                clientRequest.header("Authorization", "Basic " + request.getEncodedCredentials());
```

Comment:
I guess we should have other client authentication types support here. See `TokenClient` class.